### PR TITLE
Fix Base Class issues where setter was not available.

### PR DESCRIPTION
**Pull request by NathanMagnus**

### CODE REVIEW

File: Valigator.Core/Core/MissingGettersOrSettersException.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Valigator.Core
+{
+	public class MissingGettersOrSettersException : Exception
```
Comment (NathanMagnus) [inline Valigator.Core/Core/MissingGettersOrSettersException.cs]: New Exception type for missing Getters/Setters

Comment (JohannesMoersch) [inline Valigator.Core/Core/MissingGettersOrSettersException.cs]: Note to self: Maybe renamed to MissingAccessorsException

File: Valigator.Core/Core/Model.cs [hunk 2]
```diff
@@ -88,16 +88,63 @@ private static Expression CreatePropertyDescriptor(Expression modelExpression, P
...
+
+			if (noGetters.Any() || noSetters.Any())
+				throw new MissingGettersOrSettersException(noGetters, noSetters);
+
+			return properties;
+		}
+
+		private static IEnumerable<PropertyInfo> GetBaseProperties(Type type)
+		{
+			var currentLevelProperties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance);
```
Comment (JohannesMoersch) [inline Valigator.Core/Core/Model.cs]: Since explicit properties are technically private, should we validate private properties? I'm still unsure.

Comment (JohannesMoersch) [reply Valigator.Core/Core/Model.cs]: If we do, then we need to account for the fact that top level explicit properties will be returned by this method (if all we change is the bindingflags).

**Comment by JohannesMoersch** [comment]
Looks good. I may a couple tweaks, and then I'll merge this in.

**Pull request merged**